src/libraries/System.Private.CoreLib/src/System/Text/Unicode/Utf16Utility.Validation.cs

```diff
@@ -87,17 +88,33 @@ static Utf16Utility()
                     Vector128<short> vector8800 = Vector128.Create(unchecked((short)0x8800));
                     Vector128<ushort> vectorZero = Vector128<ushort>.Zero;
 
+
+                    Vector128<byte> initialMask = Vector128.Create((ushort)0x1001).AsByte();
```

Comment:
As I was mentioning offline, you can have a tracking issue to make this `static readonly` and see the code gen to decide which one is beneficial. You can also consider having a common method `Arm64MoveMask` that is used here and in [https://github.com/dotnet/runtime/pull/38653](https://github.com/dotnet/runtime/pull/38653).

Reply:
This case is different than `Utf8Utility`. In `Utf8Utility` you just had to check if the `mask` is zero or not and hence the implementation of `Arm64MoveMask` was correct in `Utf8Utility`. However here, you actually care about the exact mask to do further processing, so I would just stick to your original implementation in this case:

```csharp
private static readonly Vector128<byte> s_mostSignficantBitMask = Vector128.Create((byte)0x80);
private static readonly Vector128<byte> s_bitMask128 = BitConverter.IsLittleEndian ?
                                        Vector128.Create(0x80402010_08040201).AsByte() :
                                        Vector128.Create(0x01020408_10204080).AsByte();

[MethodImpl(MethodImplOptions.AggressiveInlining)]
private static int Arm64MoveMask(Vector128<byte> value)
{
    Debug.Assert(AdvSimd.Arm64.IsSupported);

    // extractedBits[i] = (value[i] & 0x80) == 0x80 & (1 << i);
    Vector128<byte> mostSignficantBitMask = s_mostSignficantBitMask;
    Vector128<byte> mostSignificantBitIsSet = AdvSimd.CompareEqual(AdvSimd.And(value, mostSignficantBitMask), mostSignficantBitMask);
    Vector128<byte> extractedBits = AdvSimd.And(mostSignificantBitIsSet, s_bitMask128);

    // self-pairwise add until all flags have moved to the first two bytes of the vector
    extractedBits = AdvSimd.Arm64.AddPairwise(extractedBits, extractedBits);
    extractedBits = AdvSimd.Arm64.AddPairwise(extractedBits, extractedBits);
    extractedBits = AdvSimd.Arm64.AddPairwise(extractedBits, extractedBits);
    return extractedBits.AsInt32().ToScalar();
}
```
In that case, my comments of type-casting and `PopCount()` will no longer be applicable.

I would also rename the method to something meaningful that captures what this method does. See my comments https://github.com/dotnet/runtime/pull/38653#discussion_r454027159.

Reply:
`Vector128.Create` is optimized to be a method local constant now, which should have better codegen than a static readonly.
Many of the existing cases using `static readonly` were working around the lack of `Vector128.Create` optimizations in netcoreapp3.1

Reply:
I think another reason of having `static readonly` I am seeing here and in other PRs is to share the value instead of replicating it everywhere.